### PR TITLE
Upgrade to .NET 8

### PR DESCRIPTION
**Pull request by millicentachieng**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoftgraph/microsoft-graph-devx-api/pull/1941)
This PR:
- Upgrade to .NET 8
- Migrate from Newtonsoft.Json to System.Text.Json

### CODE REVIEW

File: CodeSnippetsReflection.OData/LanguageGenerators/CommonGenerator.cs [hunk 1]
```diff
@@ -360,16 +349,19 @@ public static string LowerCaseFirstLetter(string s)
...
         {
             var parametersProvided = new List<string>();
-            if (!string.IsNullOrEmpty(snippetModel.RequestBody)
-                && JsonConvert.DeserializeObject(snippetModel.RequestBody) is JObject testObj)
+            if (!string.IsNullOrEmpty(snippetModel.RequestBody))
             {
-                foreach (var (key, _) in testObj)
+                using var jsonDoc = JsonDocument.Parse(snippetModel.RequestBody, JsonHelper.JsonDocumentOptions);
+                var jsonObject = jsonDoc.RootElement.Clone();
+                if (jsonDoc.RootElement.ValueKind == JsonValueKind.Object)
```
Comment (andrueastman) [inline CodeSnippetsReflection.OData/LanguageGenerators/CommonGenerator.cs:356]:
```suggestion
                var jsonObject = JsonSerializer.Deserialize<JsonElement>(snippetModel.RequestBody, JsonHelper.JsonSerializerOptions);
                if (jsonObject.ValueKind == JsonValueKind.Object)
```
Could you change to this instead accross the instances using `JsonDocument.Parse`. Prevents having to create a clone of the element which could be very big.

File: GraphWebApi/Startup.cs [hunk 2]
```diff
@@ -76,7 +69,6 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSingleton<IOpenApiService, OpenApiService>();
             services.AddSingleton<IKnownIssuesService, KnownIssuesService.Services.KnownIssuesService>();
             services.AddHttpClient<IHttpClientUtility, HttpClientUtility>();
-            services.AddControllers().AddNewtonsoftJson();
```
[the remark this answers is not in the document]
Comment (millicentachieng) [reply GraphWebApi/Startup.cs:79]: Thanks for pointing this out. I've just tested the `/samples` endpoint and enums are being serialized correctly.
image

Adding the `JsonConverter(typeof(JsonStringEnumConverter)` attribute to the property seems to be sufficient
```
[JsonPropertyName("method"), JsonRequired, JsonConverter(typeof(JsonStringEnumConverter))]
public HttpMethods Method
{
    get; set;
}
```

[the remark this answers is not in the document]
Comment (andrueastman) [reply GraphWebApi/Startup.cs:79]: Thanks for confirming. Resolving...

File: CodeSnippetsReflection.OData/LanguageGenerators/CommonGenerator.cs (hunk 1 above)
Comment (millicentachieng) [reply CodeSnippetsReflection.OData/LanguageGenerators/CommonGenerator.cs]: Thanks for the pointer. I've made the update.

**Review by andrueastman — APPROVED** [review]
👍🏼

**Pull request merged**